index.js

```diff
@@ -1,634 +1,465 @@
-	var self = this
-
-	var variables = [
-		{ label: 'Hours', name: 'hours' },
-		{ label: 'Minutes', name: 'minutes' },
-		{ label: 'Seconds', name: 'seconds' },
-		{ label: 'Name', name: 'name' },
-		{ label: 'Speed', name: 'speed' },
-		{ label: 'End Time', name: 'endTime' },
-		{ label: 'Next Timer Name', name: 'nextTimerName' },
-		{ label: 'Next Timer Duration', name: 'nextTimerDuration' },
-	]
-
-	self.setVariableDefinitions(variables)
-}
 
-instance.prototype.feedbacks = function () {
-	var self = this
-	var feedbacks = {
-		colors: {
-			label: 'Main counter Colors from timer',
-			description: 'Foreground and background colors of the timer',
-			callback: function (feedback, bank) {
-				return { color: self.fgColor, bgcolor: self.bgColor }
-			},
-		},
-		status: {
-			type: 'boolean',
-			label: 'Main counter Background Color',
-			description: 'Background color of button, based on its status',
-			options: [
-				{
-					type: 'dropdown',
-					label: 'Source',
-					id: 'Key',
-					default: 'Fullscreen',
-					choices: [
-						{ id: 'Fullscreen', label: 'Fullscreen' },
-						{ id: 'NDI', label: 'NDI' },
-						{ id: 'Message', label: 'Message' },
-						{ id: 'STM', label: 'Single Timer Mode' },
-						{ id: 'Clock', label: 'Clock' },
-						{ id: 'Pause', label: 'Pause' },
-						{ id: 'Blackout', label: 'Blackout' },
-					],
+	feedbacks() {
+		var self = this
+		var feedbacks = {
+			colors: {
+				type: 'advanced',
+				name: 'Main counter Colors from timer',
+				description: 'Foreground and background colors of the timer',
+				options: [],
+				callback: function (_feedback) {
+					return { color: self.fgColor, bgcolor: self.bgColor }
 				},
-			],
-			style: {
-				color: self.rgb(255, 255, 255),
-				bgcolor: self.rgb(255, 165, 0),
 			},
-			callback: function (feedback) {
-				return self.buttonStates[feedback.options.Key]
+			status: {
+				type: 'boolean',
+				name: 'Main counter Background Color',
+				description: 'Background color of button, based on its status',
+				options: [
+					{
+						type: 'dropdown',
+						label: 'Source',
+						id: 'Key',
+						default: 'Fullscreen',
+						choices: [
+							{ id: 'Fullscreen', label: 'Fullscreen' },
+							{ id: 'NDI', label: 'NDI' },
+							{ id: 'Message', label: 'Message' },
+							{ id: 'STM', label: 'Single Timer Mode' },
+							{ id: 'Clock', label: 'Clock' },
+							{ id: 'Pause', label: 'Pause' },
+							{ id: 'Blackout', label: 'Blackout' },
+						],
+					},
+				],
+				defaultStyle: {
+					color: combineRgb(255, 255, 255),
+					bgcolor: combineRgb(255, 165, 0),
+				},
+				callback: function (feedback) {
+					return self.buttonStates[feedback.options.Key]
+				},
 			},
-		},
-		timerName: {
-			type: 'advanced',
-			label: 'ID timer Name',
-			description: 'Timer Name with ID',
-			options: [
-				{
-					type: 'textinput',
-					label: 'ID',
-					id: 'Key',
-					default: '1',
+			timerName: {
```

Comment:
I would eventually like to see the text in this advanced feedback (and timerDuration) also setup as a variables.

Reply:
If I got you right, I think you mean giving the user the ability to show these values by using variables in addition to feedback. but the problem is these values are linked to a specific timer from the timers list with its id which is dynamically entered by the user (please check the attached image) 
In feedback, this can be fixed by passing the id in options, but I am not sure how this can be done with variables

![image](https://user-images.githubusercontent.com/12226980/226204835-34abc3e7-0b29-4e1f-b873-b80392b77a6c.png)


BTW, we already have variables for active & next timers but for the rest we used feedback

Reply:
Without putting a ton of effort/thought in, I think my understanding of your dataset would yield this type of implementation: https://github.com/bitfocus/companion-module-presentationtools-cuetimer/tree/feature/timer-vars

Reply:
It would be nice in that scenario to check the keys in `jsonData.timers` vs `self.timers` before storing to figure out if `self.variables()` is required to run.

Reply:
OK, it is easier than I thought, will merge your branch into this branch and add that keys check too, thanks!

Reply:
Please check 4639f650e66aa28c25970f3c77a9f810f1819bea and let me know if this is ok or I missed anything

Reply:
Yeah, did it test OK with the software?

Reply:
We just finished testing, sorry for the delay.  
Variables are working fine, we just removed the bg color it will only show the hex code.

So now if you think it is fine, we are ready to merge and add to the core. Thanks!